core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala

```diff
@@ -503,6 +497,62 @@ class ColumnTableTest
 
   }
 
+  test("Test DataSource API  with fully qualified table name") {
+    val tableName = "test.table1"
+    val data = Seq(Seq(1, 2, 3), Seq(7, 8, 9), Seq(9, 2, 3), Seq(4, 2, 3), Seq(5, 6, 7))
+    val rdd = sc.parallelize(data, data.length).map(s => new Data(s(0), s(1), s(2)))
+    val dataDF = snc.createDataFrame(rdd)
+    snc.createTable(tableName, "column", dataDF.schema, props)
+    dataDF.write.format("column").mode(SaveMode.Append).options(props).saveAsTable(tableName)
+    assert(snc.sql(s"select * from $tableName").collect().length == 5)
+    snc.truncateTable(tableName)
+    assert(snc.sql(s"select * from $tableName").collect().length == 0)
+    snc.dropTable(tableName)
+    println("Successful")
+  }
+
+  test("Test SQL API with fully qualified table name") {
+    val tableName = "test.table1"
+    snc.sql(s"CREATE TABLE $tableName (Col1 INT, Col2 INT, Col3 INT) USING column ")
+    assert(snc.sql("SELECT * FROM " + tableName).collect().length == 0)
+    snc.sql(s" insert into $tableName values ( 1, 2, 3)")
+    snc.sql(s" insert into $tableName values ( 2, 2, 3)")
+    snc.sql(s" insert into $tableName values ( 3, 2, 3)")
+    assert(snc.sql("SELECT * FROM " + tableName).collect().length == 3)
+    snc.sql(s"  truncate table $tableName")
+    assert(snc.sql("SELECT * FROM " + tableName).collect().length == 0)
+    snc.sql(s"DROP TABLE $tableName")
+  }
+
+  test ("Test Row buffer eviction with fully qualified table name") {
+     testRowBufferEviction("test.testTableWithSchema")
+  }
+
+  test ("Test Row buffer eviction with table name without schema") {
+    testRowBufferEviction("testTableWithoutSchema")
+  }
+
+
+  private def testRowBufferEviction(tableName:String): Unit = {
+    val props = Map(("BUCKETS" -> "1"))
+    val data = Seq(Seq(1, 2, 3), Seq(7, 8, 9), Seq(9, 2, 3), Seq(4, 2, 3), Seq(5, 6, 7))
+    val rdd = sc.parallelize(data, data.length).map(s => new Data(s(0), s(1), s(2)))
+    val dataDF = snc.createDataFrame(rdd)
+    snc.createTable(tableName, "column", dataDF.schema, props)
+    dataDF.write.format("column").mode(SaveMode.Append).options(props).saveAsTable(tableName)
+    assert(snc.sql(s"select * from $tableName").collect().length == 5)
+
+    val conn = DriverManager.getConnection("jdbc:snappydata:;query-routing=false")
+    val rs = conn.createStatement().executeQuery("select count (*) from " + tableName)
+    if (rs.next()) {
+      //The row buffer should not have more than 2 rows as the batch size is 3
+      assert(rs.getInt(1) <= 2)
+    }
+
+    rs.close()
+    conn.close()
+
+  }
```

Comment:
A test which mixes cases of with and without schema would be good. (with default APP schema)